src/pages/GameSelectionScreen.vue

```diff
@@ -385,61 +385,31 @@ export default class GameSelectionScreen extends Vue {
         }
     }
 
-    created() {
-        const self = this;
+    async created() {
         this.runningMigration = true;
-        FolderMigration.needsMigration()
-            .then(isMigrationRequired => {
-                if (!isMigrationRequired) {
-                    this.runningMigration = false;
-                } else {
-                    return FolderMigration.runMigration();
-                }
-            })
-            .then(() => {
-                this.runningMigration = false;
-            })
-            .catch((e) => {
-                console.log(e);
-                this.runningMigration = false;
-            })
-        .finally(() => {
-            ManagerSettings.getSingleton(GameManager.unsetGame()).then(settings => {
-                const lastSelectedGame = settings.getContext().global.lastSelectedGame;
-                const savedViewMode = settings.getContext().global.gameSelectionViewMode;
-                switch (savedViewMode) {
-                    case "List": this.viewMode = GameSelectionViewMode.LIST; break;
-                    case "Card":
-                    case undefined:
-                        this.viewMode = GameSelectionViewMode.CARD;
-                        break;
-                }
-                if (lastSelectedGame !== null) {
-                    const game = GameManager.gameList.find(value => value.internalFolderName === lastSelectedGame);
-                    if (game !== undefined) {
-                        this.selectedGame = game;
-                    }
-                }
-            });
-            ManagerSettings.getSingleton(GameManager.unsetGame()).then(value => {
-                this.settings = value;
-                this.favourites = value.getContext().global.favouriteGames || [];
-                if (value.getContext().global.defaultGame !== undefined) {
-                    if (value.getContext().global.defaultStore !== undefined) {
-                        const game = GameManager.gameList
-                            .find(value1 => value1.internalFolderName === value.getContext().global.defaultGame)!;
-
-                        const platform = game.storePlatformMetadata.find(value1 => value1.storePlatform === value.getContext().global.defaultStore)!;
-
-                        this.selectedGame = game;
-                        this.selectedPlatform = platform.storePlatform;
-
-                        this.proceed();
-                        return;
-                    }
-                }
-            });
-        })
+
+        try {
+            await FolderMigration.runMigration();
```

Comment:
I considered moving the migration to e.g. App.vue so it wouldn't get checked every time this view mounts (which can happen multiple times now that the app doesn't restart upon game change). But `this.runningMigration` is now used to prevent game selection when the migration is running, and handling that might become complicated if the migration is run in the App. Any thoughts where this should be called (or if it's still needed at all)?

Reply:
Could belong to Vuex? If not yet ran, run and set state.

Reply:
I'll look into it 👍🏻

Reply:
This seems to work nicely in the main Vuex store.